Dockerfile

```diff
@@ -0,0 +1,140 @@
+    && bash Mambaforge-Linux-x86_64.sh -b \
+    && rm -f Mambaforge-Linux-x86_64.sh
+RUN mamba --version
+
+# Step 2: get an up-to date cmake (HEREDOC needs DOCKER_BUILDKIT=1 enabled or build with "docker buildx")
+RUN <<-EOF
+    cmake_ubuntu_version=$(lsb_release -cs)
+    if ! wget -q --method=HEAD "https://apt.kitware.com/ubuntu/dists/$cmake_ubuntu_version/Release"; then
+      bash -c "$(wget -O - https://apt.kitware.com/kitware-archive.sh)"
+    else
+      wget -qO - https://apt.kitware.com/kitware-archive.sh | bash -s -- --release $cmake_ubuntu_version
+    fi
+    apt-get -y update
+    apt-get install -y cmake
+EOF
+
+# Step 3: dependencies for contrib etc.
+RUN apt-get install -y --no-install-recommends --no-install-suggests libsvm-dev libglpk-dev libzip-dev zlib1g-dev libxerces-c-dev libbz2-dev libomp-dev libhdf5-dev
+RUN apt-get install -y --no-install-recommends --no-install-suggests libboost-date-time1.74-dev \
+                                                                     libboost-iostreams1.74-dev \
+                                                                     libboost-regex1.74-dev \
+                                                                     libboost-math1.74-dev \
+                                                                     libboost-random1.74-dev
+RUN apt-get install -y --no-install-recommends --no-install-suggests qtbase5-dev libqt5svg5-dev libqt5opengl5-dev
+
+################################### clone OpenMS branch and the associcated contrib+thirdparties+pyOpenMS-doc submodules
+RUN git clone --recursive --depth=1 -b ${OPENMS_BRANCH} --single-branch ${OPENMS_REPO} && cd /OpenMS
+WORKDIR /OpenMS
+
+###################################
+# Step 4: Compile missing dependencies
+RUN mkdir /OpenMS/contrib-build
+WORKDIR /OpenMS/contrib-build
+
+RUN cmake -DBUILD_TYPE=EIGEN /OpenMS/contrib && rm -rf archives src
+RUN cmake -DBUILD_TYPE=COINOR /OpenMS/contrib && rm -rf archives src
+
+#################################### compile OpenMS library
+WORKDIR /
+
+# pull Linux compatible third-party dependencies and store them 
+# in directory thirdparty. Add path to them.
+RUN apt-get install -y --no-install-recommends --no-install-suggests openjdk-8-jdk
+
+WORKDIR /OpenMS
+RUN mkdir /thirdparty && \
+    git submodule update --init THIRDPARTY && \
+    cp -r THIRDPARTY/All/* /thirdparty && \
+    cp -r THIRDPARTY/Linux/64bit/* /thirdparty
+ENV PATH="/thirdparty/LuciPHOr2:/thirdparty/MSGFPlus:/thirdparty/Sirius:/thirdparty/ThermoRawFileParser:/thirdparty/Comet:/thirdparty/Fido:/thirdparty/MaRaCluster:/thirdparty/MyriMatch:/thirdparty/OMSSA:/thirdparty/Percolator:/thirdparty/SpectraST:/thirdparty/XTandem:/thirdparty/crux:${PATH}"
+
+RUN mkdir /openms-build
+WORKDIR /openms-build
+
+# configure
+RUN /bin/bash -c "cmake -DCMAKE_BUILD_TYPE='Release' -DCMAKE_PREFIX_PATH='/OpenMS/contrib-build/;/usr/;/usr/local' -DBOOST_USE_STATIC=OFF ../OpenMS"
```

Comment:
is this configuration complete? Normally this would build the Tutorials and GUI tools and documentation, which don't seem necessary for a streamlit build, right? Unless some of that is disabled by missing requirements like the Qt5GUI libs.